src/ui/pages/Menu/Menu.tsx

```diff
@@ -55,82 +99,79 @@ const Menu = () => {
     );
   };
 
-  const handleItemSelection = (index: number) => {
-    // @TODO - sdisalvo: add some logic for selection
-  };
+  const selectSubmenu = useMemo(() => {
+    // NOTE: I return empty submenu to make sure SubMenu component always render on DOM and keep animation when we open or close it
+    // Please refactor if you find out pretty solution.
+    if (selectedOption === undefined) return emptySubMenu;
 
-  const MenuItem = ({
-    index,
-    icon,
-    label,
-  }: {
-    index: number;
-    icon: string;
-    label: string;
-  }) => {
-    return (
-      <IonCol>
-        <IonCard
-          onClick={() => handleItemSelection(index)}
-          data-testid={`menu-input-item-${index}`}
-          className="menu-input"
-        >
-          <IonIcon
-            icon={icon}
-            color="primary"
-          />
-          <IonLabel>{label}</IonLabel>
-        </IonCard>
-      </IonCol>
-    );
-  };
+    const selectedSubmenu = submenuMap.get(selectedOption);
+    if (!selectedSubmenu) return emptySubMenu;
+
+    return selectedSubmenu;
+  }, [selectedOption]);
 
   return (
-    <TabLayout
-      pageId={pageId}
-      header={true}
-      title={`${i18n.t("menu.tab.header")}`}
-      additionalButtons={<AdditionalButtons />}
-    >
-      <IonGrid>
-        <IonRow className="menu-input-row">
-          <MenuItem
-            index={0}
-            icon={personCircleOutline}
-            label={`${i18n.t("menu.tab.items.profile")}`}
-          />
-          <MenuItem
-            index={1}
-            icon={walletOutline}
-            label={`${i18n.t("menu.tab.items.crypto")}`}
-          />
-        </IonRow>
-        <IonRow className="menu-input-row">
-          <MenuItem
-            index={2}
-            icon={peopleOutline}
-            label={`${i18n.t("menu.tab.items.connections")}`}
-          />
-          <MenuItem
-            index={3}
-            icon={chatbubbleOutline}
-            label={`${i18n.t("menu.tab.items.p2p")}`}
-          />
-        </IonRow>
-        <IonRow className="menu-input-row">
-          <MenuItem
-            index={4}
-            icon={fingerPrintOutline}
-            label={`${i18n.t("menu.tab.items.identity")}`}
-          />
-          <MenuItem
-            index={5}
-            icon={idCardOutline}
-            label={`${i18n.t("menu.tab.items.credentials")}`}
-          />
-        </IonRow>
-      </IonGrid>
-    </TabLayout>
+    <>
+      <TabLayout
+        pageId={pageId}
+        header={true}
+        title={`${i18n.t("menu.tab.header")}`}
+        additionalButtons={<AdditionalButtons />}
+      >
+        <IonGrid>
+          <IonRow className="menu-input-row">
+            <MenuItem
```

Comment:
@Sotatek-DukeVu Can we remove the `IonRow` and map through these elements using `MenuItem` just like we did for the Settings page?
![30042024151826](https://github.com/cardano-foundation/cf-identity-wallet/assets/119612231/69cc11cf-e290-4dab-bb4a-4006e2079c85)